{{ cookiecutter.package_name }}/setup.py

```diff
@@ -1,155 +1,33 @@
-
-# Get some values from the setup.cfg
-conf = ConfigParser()
-conf.read(['setup.cfg'])
-metadata = dict(conf.items('metadata'))
 
-PACKAGENAME = metadata.get('package_name', '{{ cookiecutter.module_name }}')
-DESCRIPTION = metadata.get('description', '{{ cookiecutter.short_description|escape }}')
-AUTHOR = metadata.get('author', '{{ cookiecutter.author_name|escape }}')
-AUTHOR_EMAIL = metadata.get('author_email', '')
-LICENSE = metadata.get('license', 'unknown')
-URL = metadata.get('url', '{{ cookiecutter.project_url }}')
-__minimum_python_version__ = metadata.get("minimum_python_version", "{{ cookiecutter.minimum_python_version }}")
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
 
-# Enforce Python version check - this is the same check as in __init__.py but
-# this one has to happen before importing ah_bootstrap.
-if sys.version_info < tuple((int(val) for val in __minimum_python_version__.split('.'))):
-    sys.stderr.write("ERROR: {{ cookiecutter.module_name }} requires Python {} or later\n".format(__minimum_python_version__))
-    sys.exit(1)
+import builtins
 
-# Import ah_bootstrap after the python version validation
+# Ensure that astropy-helpers is available
+import ah_bootstrap  # noqa
 
-import ah_bootstrap
 from setuptools import setup
-{% if cookiecutter.minimum_python_version.startswith("2") %}
-# A dirty hack to get around some early import/configurations ambiguities
-if sys.version_info[0] >= 3:
-    import builtins
-else:
-    import __builtin__ as builtins
-{%- else %}
-import builtins
-{%- endif %}
-builtins._ASTROPY_SETUP_ = True
+from setuptools.config import read_configuration
 
-from astropy_helpers.setup_helpers import (register_commands, get_debug_option,
-                                           get_package_info)
-from astropy_helpers.git_helpers import get_git_devstr
+from astropy_helpers.setup_helpers import register_commands, get_package_info
 from astropy_helpers.version_helpers import generate_version_py
 
-
-# order of priority for long_description:
-#   (1) set in setup.cfg,
-#   (2) load LONG_DESCRIPTION.rst,
-#   (3) load README.rst,
-#   (4) package docstring
-readme_glob = 'README*'
-_cfg_long_description = metadata.get('long_description', '')
-if _cfg_long_description:
-    LONG_DESCRIPTION = _cfg_long_description
-
-elif os.path.exists('LONG_DESCRIPTION.rst'):
-    with open('LONG_DESCRIPTION.rst') as f:
-        LONG_DESCRIPTION = f.read()
-
-elif len(glob.glob(readme_glob)) > 0:
-    with open(glob.glob(readme_glob)[0]) as f:
-        LONG_DESCRIPTION = f.read()
-
-else:
-    # Get the long description from the package's docstring
-    __import__(PACKAGENAME)
-    package = sys.modules[PACKAGENAME]
-    LONG_DESCRIPTION = package.__doc__
-
 # Store the package name in a built-in variable so it's easy
 # to get from other parts of the setup infrastructure
-builtins._ASTROPY_PACKAGE_NAME_ = PACKAGENAME
+builtins._ASTROPY_PACKAGE_NAME_ = read_configuration('setup.cfg')['metadata']['name']
```

Comment:
ooi, where is this used?

Reply:
I can't find it in the code of astropy_helpers or core by searching on github fwiw.

Reply:
it's used in ``setup_pacakge.py`` for the package template but we could arguably move the definition of the package data to ``setup.cfg``

Reply:
Where is it used? Can you link me?

Reply:
It's no longer in the template but it's used in a loooot of packages:

https://github.com/search?q=_ASTROPY_PACKAGE_NAME_&type=Code

I guess there's no reason we can't remove it though. Maybe we can even do batch PRs to update all the packages to hard-code the package name instead.

Reply:
For now I'm leaving this in though.

Reply:
>  Maybe we can even do batch PRs to update all the packages to hard-code the package name instead.

👍 to that

Reply:
(but I think we may even want to batchPR the uncontroversial things from the rendered version after we tag a new version here.)